mmedit/datasets/pipelines/augmentation.py

```diff
@@ -145,6 +148,190 @@ def __repr__(self):
         return repr_str
 
 
+@PIPELINES.register_module()
+class RandomResizedCrop(object):
+    """Crop the given numpy array to random size and aspect ratio.
+
+    A crop of random size (default: of 0.08 to 1.0) of the original size and a
+    random aspect ratio (default: of 3/4 to 4/3) of the original aspect ratio
+    is made. This crop is finally resized to given size.
+
+    This code is partially adopted from
+    torchvision.transforms.RandomResizedCrop:
+    [https://pytorch.org/vision/stable/_modules/torchvision/transforms/\
+        transforms.html#RandomResizedCrop].
+
+    Args:
+        keys (list[str]): The images to be resized and random-cropped.
+        crop_size (int | tuple[int]): Target spatial size (h, w).
+        scale (tuple[float]): Range of size of the origin size cropped.
+                              Default: (0.08, 1.0)
+        ratio (tuple[float]): Range of aspect ratio of the origin aspect ratio
+                              cropped. Default (3./4., 4./3.)
+        interpolation (str): Algorithm used for interpolation:
+            "nearest" | "bilinear" | "bicubic" | "area" | "lanczos".
+            Default: "bilinear".
+    """
+
+    def __init__(self,
+                 keys,
+                 crop_size,
+                 scale=(0.08, 1.0),
+                 ratio=(3. / 4., 4. / 3.),
+                 interpolation='bilinear'):
+        assert keys, 'Keys should not be empty.'
+        if mmcv.is_tuple_of(crop_size, int):
+            crop_size = crop_size
+        elif isinstance(crop_size, int):
+            crop_size = (crop_size, crop_size)
+        else:
+            raise TypeError('Elements of crop_size must be int and crop_size '
+                            'must be either tuple or int, but got'
+                            f'{type(crop_size[0])} in {type(crop_size)}')
+        if not mmcv.is_tuple_of(scale, float):
+            raise TypeError('Elements of scale must be float and scale '
+                            'must be tuple, but got '
+                            f'{type(scale[0])} in {type(scale)}')
+        if not mmcv.is_tuple_of(ratio, float):
+            raise TypeError(
+                'Elements of ratio must be float and ratio must be'
+                f' tuple, but got {type(ratio[0])} in {type(ratio)}')
+
+        self.keys = keys
+        self.crop_size = crop_size
+        self.scale = scale
+        self.ratio = ratio
+        self.interpolation = interpolation
+
+    @staticmethod
+    def get_params(data, scale, ratio):
```

Comment:
We can simply remove `@staticmethod` and use `def get_params(self):`

In this case, we can directly use `self.scale` and `self.ratio`